components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/AuthorizationHandler.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.management.apis.security.handler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.wso2.micro.core.util.CarbonException;
+import org.wso2.micro.integrator.management.apis.Constants;
+import org.wso2.micro.integrator.management.apis.ManagementApiUndefinedException;
+import org.wso2.micro.integrator.security.MicroIntegratorSecurityUtils;
+import org.wso2.micro.integrator.security.user.api.UserStoreException;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.IOException;
+import java.util.Arrays;
+
+import static org.wso2.micro.integrator.management.apis.Constants.USERNAME_PROPERTY;
+
+public class AuthorizationHandler extends SecurityHandlerAdapter {
+
+    private static final Log LOG = LogFactory.getLog(AuthorizationHandler.class);
+    private String name;
+    private MessageContext messageContext;
+
+    public AuthorizationHandler() throws CarbonException, XMLStreamException, IOException, ManagementApiUndefinedException {
+        super();
+    }
+
+    @Override
+    public Boolean invoke(MessageContext messageContext) {
+
+        this.messageContext = messageContext;
+        return super.invoke(messageContext);
+    }
+
+    @Override
+    public String getName() {
+
+        return this.name;
+    }
+
+    @Override
+    public void setName(String name) {
+
+        this.name = name;
+    }
+
+    @Override
+    protected Boolean authenticate(String authHeaderToken) {
+
+        if ((messageContext.getTo().getAddress()).startsWith((Constants.REST_API_CONTEXT + Constants.PREFIX_USERS))) {
```

Comment:
Can we get "(Constants.REST_API_CONTEXT + Constants.PREFIX_USERS)" as a parameter for the handler from internal api xml, then we may use for other resources as well

Reply:
I thought about it, but didn't do since at the moment we do not have any other resources that require this capability. When, we get another let's do this.

Reply:
Changing is the state of message processor is kind of write operation and people might tend to protect it .

Reply:
done